compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -394,7 +394,10 @@
             { ParserRuleContext.OPEN_PARENTHESIS, ParserRuleContext.LT };
 
     private static final ParserRuleContext[] FUNCTION_KEYWORD_RHS =
-            { ParserRuleContext.FUNC_NAME, ParserRuleContext.OPEN_PARENTHESIS };
+            { ParserRuleContext.FUNC_NAME, ParserRuleContext.OPEN_PARENTHESIS, ParserRuleContext.FUNC_TYPEDESC_RHS };
+
+    private static final ParserRuleContext[] FUNC_NAME_RHS =
+            { ParserRuleContext.OPEN_PARENTHESIS, ParserRuleContext.ASSIGN_OP, ParserRuleContext.SEMICOLON };
```

Comment:
Since every case is handled via `FUNCTION_KEYWORD_RHS` I think we don't need this

Reply:
Removed.